src/Snap/Internal/Http/Server/Parser.hs

```diff
@@ -173,18 +177,35 @@ parseRequest input = do
                     in (Just $! host, uri)
               | otherwise = (Nothing, s)
 
-    pVer s = if "HTTP/" `S.isPrefixOf` s
-               then pVers (S.unsafeDrop 5 s)
-               else (1, 0)
+    -- RFC 7230 section 2.6 defines
+    --
+    --   HTTP-version = HTTP-name "/" DIGIT "." DIGIT
+    --   HTTP-name    = %x48.54.54.50 ; "HTTP", case-sensitive
+    --
+    pVer s = case bsStripHttpPrefix s of
```

Comment:
This code is in the core accept loop and so we can't tolerate code that's significantly slower here. E.g. `bsStripHttpPrefix` can't return `Maybe` because that's an extra allocation you can't afford; instead you would need to return null string as your empty case.